src/main/resources/templates/macros/report/output.vm

```diff
@@ -1,20 +1,49 @@
-#macro(includeOutput, $output)
+#macro(includeOutput, $outputs)
 
-#if ($output)
-  <div class="outputs inner-level">
-    #foreach($message in $output.getMessages())
+#set($combineOutputs = "true")
+#foreach($output in $step.getOutput())
+  #if(${output.getMessages().size()} > 1)
+    #set($combineOutputs = "false")
+  #end
+#end
+
+#if($step.getOutput())
+  #if($combineOutputs == "true")
+    <div class="outputs inner-level">
       <div class="output indention">
         #set($outputId = $counter.next())
-        <div data-toggle="collapse" class="collapsable-control" data-target="#msg-$outputId">
-          #set($index = $foreach.index + 1)
-          <a>Output $index</a>
+        <div data-toggle="collapse" class="collapsable-control" data-target="#output-$outputId">
+          <a>Output</a>
         </div>
-        <div id="msg-$outputId" class="collapse collapsable-details">
-          <pre>$message</pre>
+        <div id="output-$outputId" class="collapse collapsable-details">
+          <pre class="white-space-normal">
+            #foreach($output in $outputs)
+              <span>$output.getMessages()[0]</span><br>
+            #end
+          </pre>
+        </div>
+      </div>
+    </div>
+  #else
+    #foreach($output in $outputs)
```

Comment:
again - you iterate over outputs two times - can you merge into single loop? Right now I'm not sure what is going on here - this code is quite complex :(